lib/ohai/mixin/rackspace_metadata.rb

```diff
@@ -0,0 +1,51 @@
+#
+# Author:: Alexey Karpik <alexey.karpik@rightscale.com>
+# Author:: Cary Penniman (<cary@rightscale.com>)
+# Author:: Peter Schroeter <peter.schroeter@rightscale.com>
+# Author:: Stas Turlo <stanislav.turlo@rightscale.com>
+# Copyright:: Copyright (c) 2010-2014 RightScale Inc
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "resolv"
+require "json"
+
+module ::Ohai::Mixin::RackspaceMetadata
+  def on_windows?
+    RUBY_PLATFORM =~ /windows|cygwin|mswin|mingw|bccwin|wince|emx/
+  end
+
+  def xenstore_command(command, args)
+    if on_windows?
+      xen_store_client_path = '"c:\Program Files\Citrix\XenTools\xenstore_client.exe"'
```

Comment:
The hard-coded path won't work -- for example, Windows can be (and often is, especially in datacenters) installed without a C drive. Also, any software can be installed to an arbitrary location. Administrators control that through Windows Installer / Group Policy / SCCM, etc. So there needs to be a different way of finding this command -- if it's in the path, just specifying the executable should work.
A more reliable way would be to look for a registry entry created by this software -- those are canonical and can't be configured by users. Ruby has helper code to read the registry: http://ruby-doc.org/stdlib-2.0.0/libdoc/fiddle/rdoc/Win32/Registry/API.html.